src/framework/components/element/component.js

```diff
@@ -1044,26 +1002,753 @@ class ElementComponent extends Component {
+     *
+     * @type {Vec2}
+     */
+    set alignment(arg) {
+        this._setValue('alignment', arg);
+    }
+
+    get alignment() {
+        if (this._text) {
+            return this._text.alignment;
+        }
+
+        return null;
+    }
+
+    /**
+     * Automatically set the width of the component to be the same as the textWidth. Only works for
+     * {@link ELEMENTTYPE_TEXT} types.
+     *
+     * @type {boolean}
+     */
+    set autoWidth(arg) {
+        this._setValue('autoWidth', arg);
+    }
+
+    get autoWidth() {
+        if (this._text) {
+            return this._text.autoWidth;
+        }
+
+        return null;
+    }
+
+    /**
+     * Automatically set the height of the component to be the same as the textHeight. Only works
+     * for {@link ELEMENTTYPE_TEXT} types.
+     *
+     * @type {boolean}
+     */
+    set autoHeight(arg) {
+        this._setValue('autoHeight', arg);
+    }
+
+    get autoHeight() {
+        if (this._text) {
+            return this._text.autoHeight;
+        }
+
+        return null;
+    }
+
+    /**
+     * Reorder the text for RTL languages using a function registered by
+     * `app.systems.element.registerUnicodeConverter`.
+     *
+     * @type {boolean}
+     */
+    set rtlReorder(arg) {
+        this._setValue('rtlReorder', arg);
+    }
+
+    get rtlReorder() {
+        if (this._text) {
+            return this._text.rtlReorder;
+        }
+
+        return null;
+    }
+
+    /**
+     * Convert unicode characters using a function registered by `app.systems.element.registerUnicodeConverter`.
+     *
+     * @type {boolean}
+     */
+    set unicodeConverter(arg) {
+        this._setValue('unicodeConverter', arg);
+    }
+
+    get unicodeConverter() {
+        if (this._text) {
+            return this._text.unicodeConverter;
+        }
+
+        return null;
+    }
+
+    /**
+     * The text to render. Only works for {@link ELEMENTTYPE_TEXT} types. To override certain text
+     * styling properties on a per-character basis, the text can optionally include markup tags
+     * contained within square brackets. Supported tags are:
+     * 1.  `color ` - override the element's  `color ` property.
+     * Examples:
+     *      -  `[color="#ff0000"]red text[/color] `
+     *      -  `[color="#00ff00"]green text[/color] `
+     *      -  `[color="#0000ff"]blue text[/color] `
+     * 2.  `outline ` - override the element's  `outlineColor ` and  `outlineThickness ` properties.
+     * Example:
+     *      -  `[outline color="#ffffff" thickness="0.5"]text[/outline] `
+     * 3.  `shadow ` - override the element's  `shadowColor ` and  `shadowOffset ` properties.
+     * Examples:
+     *      -  `[shadow color="#ffffff" offset="0.5"]text[/shadow] `
+     *      -  `[shadow color="#000000" offsetX="0.1" offsetY="0.2"]text[/shadow]
+     *
+     * ` Note that markup tags are only processed if the text element's  `enableMarkup ` property is
+     * set to true.
```

Comment:
This did not translate over from the original `@property` formatting correctly. Compare against the original...

Reply:
Fixed